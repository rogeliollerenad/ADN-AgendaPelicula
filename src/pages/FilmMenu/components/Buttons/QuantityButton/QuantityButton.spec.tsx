import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { QuantityButton } from '.';

describe('QuantityButton tests', () => {
  let onQuantityChange: jest.Mock;
  let quantity: number;

  beforeEach(() => {
    onQuantityChange = jest.fn();
    quantity = 1;
  });

  it('should match snapshot', () => {
    const { container } = render(
      <QuantityButton
        onQuantityChange={onQuantityChange}
        quantity={quantity}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should add quantity', async () => {
    const expectedQuantity = 1;
    const { findByTestId } = render(
      <QuantityButton
        onQuantityChange={onQuantityChange}
        quantity={quantity}
      />,
    );

    const addButton = await findByTestId('add');

    fireEvent.click(addButton);

    expect(onQuantityChange).toHaveBeenCalledWith(expectedQuantity);
  });

  it('should rest quantity', async () => {
    const expectedQuantity = -1;
    const { findByTestId } = render(
      <QuantityButton
        onQuantityChange={onQuantityChange}
        quantity={quantity}
      />,
    );

    const addButton = await findByTestId('rest');

    fireEvent.click(addButton);

    expect(onQuantityChange).toHaveBeenCalledWith(expectedQuantity);
  });
});
