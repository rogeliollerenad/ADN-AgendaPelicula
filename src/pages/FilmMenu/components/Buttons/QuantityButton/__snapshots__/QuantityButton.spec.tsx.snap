// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityButton tests should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-unitsContainer-1"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root"
      data-testid="rest"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13H5v-2h14v2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      1
    </p>
    <button
      class="MuiButtonBase-root MuiIconButton-root"
      data-testid="add"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
